software/soapysdr-xtrx/Streaming.cpp

```diff
@@ -339,29 +351,33 @@ void SoapyXTRX::releaseWriteBuffer(SoapySDR::Stream */*stream*/, size_t handle,
     checked_ioctl(_fd, LITEPCIE_IOCTL_MMAP_DMA_READER_UPDATE, &mmap_dma_update);
 }
 
-void deinterleave(const int8_t *src, void *dst, uint32_t len, std::string format, size_t offset)
+void deinterleave(const void *src, size_t src_offset, void *dst, size_t dst_offset,
+                  size_t len, std::string format, size_t channel)
 {
     if (format == SOAPY_SDR_CS16) {
-        int16_t *samples_cs16 = (int16_t *)dst + offset * BYTES_PER_SAMPLE;
-        for (uint32_t i = offset; i < len; i += 2)
+        int16_t *src_cs16 = (int16_t *)src + 4*src_offset;
+        int16_t *dst_cs16 = (int16_t *)dst + 2*dst_offset;
+        for (uint32_t i = 0; i < len; i += 1)
         {
-            samples_cs16[i * BYTES_PER_SAMPLE] = (int16_t)(src[i * BYTES_PER_SAMPLE] << 8);
-            samples_cs16[i * BYTES_PER_SAMPLE + 1] = (int16_t)(src[i * BYTES_PER_SAMPLE + 1] << 8);
+            dst_cs16[2*i]     = src_cs16[4*i + channel];
+            dst_cs16[2*i + 1] = src_cs16[4*i + channel + 2];
```

Comment:
```suggestion
            dst_cs16[2*i + 1] = src_cs16[4*i + channel + 1];
```
I'm pretty sure we want `+1` here, since the data format is `Ai Aq Bi Bq` where `Ai` refers to the real part of channel A.  You can verify this by opening up LimeSuiteGUI, going to `LimeLight & PAD` tab, going to the `Sample position & Clock` sub-tab, and looking at the sample position configuration.

Reply:
I followed what Steve said here: https://github.com/JuliaComputing/xtrx_julia/pull/59#issue-1359387234

Reply:
Yep, sorry, we disproved that comment but never wrote it down anywhere.